crates/oxc_linter/src/rules/eslint/no_unused_vars/usage.rs

```diff
@@ -610,6 +610,9 @@ impl<'a> Symbol<'_, 'a> {
                     if matches!(parent, AstKind::CallExpression(_)) {
                         continue;
                     }
+                    if matches!(parent, AstKind::AwaitExpression(_) | AstKind::YieldExpression(_)) {
+                        continue;
+                    }
```

Comment:
```suggestion
                    if matches!(parent, AstKind::CallExpression(_) | AstKind::AwaitExpression(_) | AstKind::YieldExpression(_)) {
                        continue;
                    }
```

These two branches can be folded into one.